SimPEG/regularization/base.py

```diff
@@ -1279,17 +1279,17 @@ def f_m(self, m):
         .. math::
             \phi_m (\mathbf{m}) = \frac{1}{2} \Big \| \mathbf{W \, f_m} \Big \|^2
         """
-        dfm_dl = self.cell_gradient @ (self.mapping * self._delta_m(m))
+        dfm_dl = self.mapping * self._delta_m(m)
 
         if self.units is not None and self.units.lower() == "radian":
-            dfm_dl = (
-                utils.mat_utils.coterminal(dfm_dl * self.length_scales)
+            return self.cell_gradient.T @ (
```

Comment:
haven't worked through it in detail, but something looks odd here-- in this block you only multiply by the cell_gradient once before returning, but if you take the other branch you do it twice. I assume you're fixing a problem with how the gradient was implemented before, but can you explain it to me?

Reply:
(happy to talk through it on Monday)

Reply:
it looks like it is multiplied by gradient twice:
1. coterminal(`self.cell_gradient.sign() @ dfm_dl) / self.length_scales ` (the idea here is to apply coterminal to the angle difference, before dividing by the distance)
2. self.cell_gradient.T @ ...

This comes from this PR: https://github.com/simpeg/simpeg/pull/1299

Also, we are not using second derivative smoothness anywhere.

Reply:
i still don't quite follow but if it's in simpeg main i'm not going to worry to much about it for now